File: Java/pattern/VisitorProj/src/com/pattern/ConcreteNode2.java
package com.pattern;

public class ConcreteNode2 implements Node {

	@Override
	public void showMe() {
		System.out.println("ConcreteNode2::showMe()");

	}

	@Override
	public void accept(Visitor v) {
		v.visit(this);
		
	}

}

File: cplusplus/design_pattern/VisitorProj/ConcreteNode2.cpp
#include "stdafx.h"
#include "ConcreteNode2.h"
#include "Visitor.h"

ConcreteNode2::ConcreteNode2()
{
}


ConcreteNode2::~ConcreteNode2()
{
}

void ConcreteNode2::showMe()
{
    std::cout << "ConcreteNode2::showMe()" << std::endl;
}

void ConcreteNode2::accept(std::shared_ptr<Visitor> v)
{
    v->visit(shared_from_this());
}
